codalab/worker/state_committer.py

```diff
@@ -26,7 +28,14 @@ def load(self, default=None):
 
     def commit(self, state):
         """ Write out the state in JSON format to a temporary file and rename it into place """
-        with tempfile.NamedTemporaryFile() as f:
-            f.write(pyjson.dumps(state).encode())
-            f.flush()
-            shutil.copyfile(f.name, self._state_file)
+        with tempfile.NamedTemporaryFile(delete=False) as f:
```

Comment:
Since we are setting `delete=False`, can you add a test that ensures that the temp file eventually gets deleted?